docs/Adding-Software-to-Homebrew.md

```diff
@@ -8,46 +8,139 @@ Before you start, please check the open pull requests for [Homebrew/homebrew-cor
+* make sure it can be accepted by checking [Acceptable Casks](Acceptable-Casks.md).
 * check that the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
+* if you are just updating the version of an existing cask, see the streamlined method in ['submit a new version of an existing cask'](How-To-Open-a-Homebrew-Pull-Request.md#submit-a-new-version-of-an-existing-cask).
 
 ### Writing the cask
 
-Making a new cask is easy. Follow the directions in [How to Open a Homebrew Pull Request](How-To-Open-a-Homebrew-Pull-Request.md#cask-related-pull-request) to begin.
+Making a new cask is easy, and the [Cask Cookbook](Cask-Cookbook.md) is an essential resource to guide you through the process. The Cookbook provides detailed instructions, best practices, and solutions to common problems you may encounter when creating a cask. Familiarizing yourself with its contents before you start is advisable.
+
+1. Begin by researching existing casks in Homebrew that are similar to the software you plan to add. This will help you understand the typical structure and conventions used in casks. Start by tapping `homebrew/cask`: run `brew tap homebrew/cask` to clone the `homebrew/cask` tap to the path returned by `brew --repository homebrew/cask`.
+
+1. If you're starting from scratch, you can use the [`brew create --cask <download-url>` command](Manpage.md#create-options-url) to produce a basic version of your cask.
+
+  After executing the `create` command, `EDITOR` will open with a cask template at `$(brew --repository homebrew/cask)/Casks/<prefix>/<cask>.rb` (where `<prefix>` is the first letter of your cask's name and `<cask>` is the name of your cask). The template will appear as follows:
+
+  ```ruby
+  cask "my-new-cask" do
+    version ""
+    sha256 ""
+  
+    url "download-url"
+    name ""
+    desc ""
+    homepage ""
+  
+    app ""
+  end
+  ```
+
+1. You will now have to develop the boilerplate code from `brew create --cask` into a complete cask. Your main references will be the [Cask Cookbook](Cask-Cookbook.md), similar existing casks, and the official documentation of the software you're packaging.
+
+   For further specifics and additional insights, refer to ['Additional Cask Details and Examples'](#additional-cask-details-and-examples).
+
+1. It's important to include a good test in your cask. Since the Cask Cookbook doesn't provide specific guidance on this, you can refer to the [Add a test to the formula](Formula-Cookbook.md#add-a-test-to-the-formula) section in the Formula Cookbook for assistance.
+
+### Testing and auditing the Cask
+
+1. Test your cask installation using `brew install --cask $(brew --repository homebrew/cask)/Casks/<prefix>/<cask>.rb`, where `<prefix>` is the first letter of your cask's name and `<cask>` is the name of your cask. Correct any errors that occur and repeat the installation until it completes without errors.
```

Comment:
Same as for the formula.

Reply:
See my comment there:

- https://github.com/Homebrew/brew/pull/16327#discussion_r1425291768

Reply:
@MikeMcQuaid Fixed in f934604734eb6a78230ea0b76ac2cbce4c6a3551

I'll leave this comment thread open for visibility for now; but once you check it and are happy, leave a comment and then feel free to mark it resolved.